### ISSUE TITLE
Detect uncalled function statements

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#47698]
# Suggestion

## 🔍 Search Terms
- uncalled function
- check
- statement

## ⭐ Suggestion

Today TS can detect uncalled functions that appear in conditionals. I propose also extending this to uncalled functions that appear as statements

## 📃 Motivating Example

```ts
function doImportantStuff() { }

doImportantStuff;  // Mistake here. The user forgot to call the func

if (doImportantStuff) { // This one is detected today

}
```

## 💻 Use Cases

This can help catch a common programming mistake. As long as the function being checked has a `const` binding and is always a function type, I don't think there are cases where you actually want to write `doImportantStuff;` as a statement
